data_structure/segment_trees.cpp

```diff
@@ -0,0 +1,92 @@
+/* Program describes usage of a popular Data Structure: Segment Trees
+ *  in Finding maximum number and its index on a given Interval
+ *  in logarithmic = Log(N) Time using SegmentTree Data Structures
+ *  with PreProcessing O(N * Log(N)) Time
+ */
+
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
+
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <vector>
+
+// Each Node contains it's value and index
```

Comment:
Please use a comment structure like:

```c++
/**
 * Something here
 */
```
**except** inside functions.
Same in other parts of the code.